config/main.py

```diff
@@ -2064,6 +2064,538 @@ def delete_snmptrap_server(ctx, ver):
+        click.echo("Bad SNMP user")
+        sys.exit(1)
+    user_type = user_type.lower()
+    if not is_valid_user_type(user_type):
+        click.echo("User type {} is invalid.  Must be noauthnopriv, authnopriv, or priv".format(user_type))
+        sys.exit(2)
+    if user_type == "noauthnopriv":
+        user_type = "noAuthNoPriv"
+    elif user_type == "authnopriv":
+        user_type = "AuthNoPriv"
+    elif user_type == "priv":
+        user_type = "Priv"
+    user_permission_type = user_permission_type.upper()
+    if not is_valid_community_type(user_permission_type):
+        click.echo("User permission type {} is invalid.  Must be RO or RW".format(user_permission_type))
+        sys.exit(3)
+    if user_type == "noAuthNoPriv":
+        if user_auth_type:
+            click.echo("User auth type not used with 'noAuthNoPriv'.  Please use 'AuthNoPriv' or 'Priv' instead")
+            sys.exit(4)
+    if user_type is not 'noAuthNoPriv':
+        if not user_auth_type:
+            click.echo("User auth type is missing.  Must be MD5, SHA, or HMAC-SHA-2")
+            sys.exit(5)
+        if user_auth_type:
+            user_auth_type = user_auth_type.upper()
+            if not is_valid_auth_type(user_auth_type):
+                click.echo("User auth type {} is invalid.  Must be MD5, SHA, or HMAC-SHA-2".format(user_auth_type))
+                sys.exit(6)
+            elif not user_auth_password:
+                click.echo("User auth password is missing")
+                sys.exit(7)
+            elif user_auth_password:
+                if not snmp_user_secret_check(user_auth_password):
+                    click.echo("Bad user auth password")
+                    sys.exit(8)
+        if user_type == "AuthNoPriv":
+            if user_encrypt_type:
+                click.echo("User encrypt type not used with 'AuthNoPriv'.  Please use 'Priv' instead")
+                sys.exit(9)
+        elif user_type == "Priv":
+            if not user_encrypt_type:
+                click.echo("User encrypt type is missing.  Must be DES or AES")
+                sys.exit(10)
+            if user_encrypt_type:
+                user_encrypt_type = user_encrypt_type.upper()
+                if not is_valid_encrypt_type(user_encrypt_type):
+                    click.echo("User encryption type {} is invalid.  Must be DES or AES".format(user_encrypt_type))
+                    sys.exit(11)
+                elif not user_encrypt_password:
+                    click.echo("User encrypt password is missing")
+                    sys.exit(12)
+                elif user_encrypt_password:
+                    if not snmp_user_secret_check(user_encrypt_password):
+                        click.echo("Bad user encrypt password")
+                        sys.exit(13)
+    snmp_users = db.cfgdb.get_table("SNMP_USER")
+    if user in snmp_users.keys():
+        click.echo("SNMP user {} is already configured".format(user))
+        sys.exit(14)
```

Comment:
>14 [](http://example.com/codeflow?start=17&length=2)

Better to define an Enum class to represent all the possible return codes.

Reply:
Updated to user Enum.